version: "3.9"

services:
  db:
    image: postgres
    volumes:
      - ./docker_data/db:/var/lib/postgresql/data
    env_file:
      - postgres.env
    healthcheck:
      test: "pg_isready --username=postgres && psql --username=postgres --list"
      timeout: 10s
      retries: 20
  web:
    image: django_playground:latest
    build: .
    command: sh -c "/home/app/entrypoint.sh"
    volumes:
      - .:/home/app/django
    ports:
      - "81:81"
    env_file:
      - django.env
    depends_on:
      db:
        condition: service_healthy
      elasticsearch:
        condition: service_started
      redis:
        condition: service_started
  elasticsearch:
    image: docker.elastic.co/elasticsearch/elasticsearch:8.2.2
    environment:
      - xpack.security.enabled=false
      - discovery.type=single-node
      - ES_JAVA_OPTS=-Xms1g -Xmx1g
    volumes:
      - ./docker_data/elasticsearch:/usr/share/elasticsearch/data
  redis:
    image: redis:alpine
    volumes:
      - ./docker_data/redis:/data
    ports:
      - "6500:6379"
  celery:
    image: django_playground:latest
    command: celery -A project worker -l info -E
    volumes:
      - .:/home/app/django
    env_file:
      - django.env
    depends_on:
      - redis
      - db